### PR TITLE
support absolute paths and file:// extension

### LINKED ISSUE #4316: sourcemaps always forced to be relative even for absolute paths and http/https/file locations
"parcel-bundler": "^1.12.4",

In parcel-bundler/src/utils/loadSourceMap.js The code always assumes that source maps are relative to the asset it is describing, which is not always the case.

For example, if the source map sources item starts with "/" it should probably not use the relative scheme. Source maps that start with http/https should be passed through and those with file: could be passed through or resolved (parcel could choose or perhaps a switch).

For example, this fixes the absolute path issue:

```js
     if(source.startsWith("file:///")) source = source.slice(7) // triple slash is also absolute
    try {
              let sourceFile = path.isAbsolute(source) ? // CHECK FOR ABSOLUTE PATH HERE
                source:
                path.join(
                path.dirname(filename),
                sourceMap.sourceRoot || '',
                source
              );
```

Is this something that makes sense to fix?

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

This PR implements the removal of `file://` and support for absolute paths in sourcemap comments.

This does seem to be a very niche edge case but apprarently it happens so this PR implements it.

Fixes #4316

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/utils/src/sourcemap.js [hunk 1]
```diff
@@ -30,15 +30,18 @@ export async function loadSourceMapUrl(
   if (match) {
     let url = match[1].trim();
     let dataURLMatch = url.match(DATA_URL_RE);
-    filename = dataURLMatch ? filename : path.join(path.dirname(filename), url);
+    let mapFilePath = dataURLMatch ? filename : url.replace('file://', '');
```
Comment (mischnic) [inline packages/core/utils/src/sourcemap.js:33]:
```suggestion
    let mapFilePath = dataURLMatch ? filename : url.replace(/^file:///, '');
```

or something along those lines?

**Comment by mischnic** [comment]
So absolute paths are treaded as relative?

Just out of curiosity: do you have an example of a package or of a tool that emits these maps?

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.68s | -196.00ms |
| Cached | 1.51s | -44.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 182.00ms | +133.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 182.00ms | +132.00ms ⚠️ |
| dist/legacy/index.45cccf70.js | 1.10kb | +0.00b | 2.23s | -164.00ms 🚀 |
| dist/modern/index.1964d1f6.js | 1.10kb | +0.00b | 2.23s | -163.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 1.70s | +137.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 1.69s | +140.00ms ⚠️ |
| dist/legacy/index.b3acc8eb.css | 77.00b | +0.00b | 1.49s | +99.00ms ⚠️ |
| dist/modern/index.7bc22a8b.css | 77.00b | +0.00b | 2.22s | -164.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 32.00ms | +11.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 32.00ms | +11.00ms ⚠️ |
| dist/legacy/index.45cccf70.js | 1.10kb | +0.00b | 169.00ms | +77.00ms ⚠️ |
| dist/modern/index.1964d1f6.js | 1.10kb | +0.00b | 160.00ms | +69.00ms ⚠️ |
| dist/legacy/index.html | 701.00b | +0.00b | 182.00ms | +77.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 160.00ms | +69.00ms ⚠️ |
| dist/legacy/index.b3acc8eb.css | 77.00b | +0.00b | 169.00ms | +77.00ms ⚠️ |
| dist/modern/index.7bc22a8b.css | 77.00b | +0.00b | 160.00ms | +69.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 30.88s | -3.02s 🚀 |
| Cached | 1.69s | +53.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/PermalinkedComment.9c5ad065.js | 4.22kb | +0.00b | 6.27s | -6.37s 🚀 |
| dist/UserProfile.c1927888.js | 1.62kb | +0.00b | 6.27s | -6.37s 🚀 |
| dist/NotFound.f8a0616d.js | 387.00b | +0.00b | 6.27s | -6.37s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 478.68kb | +0.00b | 129.00ms | +48.00ms ⚠️ |
| dist/PermalinkedComment.9c5ad065.js | 4.22kb | +0.00b | 112.00ms | +39.00ms ⚠️ |
| dist/UserProfile.c1927888.js | 1.62kb | +0.00b | 112.00ms | +39.00ms ⚠️ |
| dist/NotFound.f8a0616d.js | 387.00b | +0.00b | 112.00ms | +38.00ms ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 18.00ms | +4.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 3.31m | +1.53s |
| Cached | 4.86s | +62.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.fa25440a.js | 2.72mb | +3.00b ⚠️ | 1.81m | +1.35s |
| dist/media-viewer.bcba38b3.js | 71.96kb | +0.00b | 17.26s | -938.00ms 🚀 |
| dist/card.0313bac8.js | 52.78kb | +0.00b | 1.82m | +44.00s ⚠️ |
| dist/Modal.19ffed32.js | 41.27kb | +0.00b | 7.54s | -1.01s 🚀 |
| dist/component.d7c29980.js | 30.70kb | +0.00b | 3.84s | -1.29s 🚀 |
| dist/component.54080a26.js | 22.50kb | +0.00b | 7.53s | -1.02s 🚀 |
| dist/dropzone.7c704bdb.js | 17.27kb | +0.00b | 1.81m | +43.51s ⚠️ |
| dist/smartMediaEditor.9623570c.js | 16.87kb | +0.00b | 1.06m | -43.51s 🚀 |
| dist/js.5733a017.js | 16.40kb | +0.00b | 7.53s | -1.02s 🚀 |
| dist/EmojiTypeAheadComponent.1990d771.js | 8.67kb | +0.00b | 1.02m | +47.91s ⚠️ |
| dist/component.77c0f682.js | 6.05kb | +0.00b | 7.97s | -580.00ms 🚀 |
| dist/card.383bd9bd.js | 5.61kb | +0.00b | 17.25s | -947.00ms 🚀 |
| dist/media-viewer.59c17dad.js | 3.92kb | +0.00b | 17.25s | -946.00ms 🚀 |
| dist/index.ac562c80.css | 3.46kb | +0.00b | 7.67s | -1.00m 🚀 |
| dist/png-chunks-extract.a844fdb6.js | 3.41kb | +0.00b | 7.53s | -1.02s 🚀 |
| dist/Modal.0304b9ed.js | 3.08kb | +0.00b | 7.53s | +620.00ms ⚠️ |
| dist/16.290b4a43.js | 2.38kb | +0.00b | 3.83s | -1.29s 🚀 |
| dist/EmojiUploadComponent.d54a1fc6.js | 2.18kb | +0.00b | 1.02m | +47.91s ⚠️ |
| dist/ResourcedEmojiComponent.abe8596c.js | 2.00kb | +0.00b | 7.93s | +1.95s ⚠️ |
| dist/date.231500d2.js | 1.73kb | +0.00b | 8.55s | -573.00ms 🚀 |
| dist/16.d16f14ed.js | 1.73kb | +0.00b | 8.23s | -531.00ms 🚀 |
| dist/images.80c92f5e.js | 1.67kb | +0.00b | 8.55s | -677.00ms 🚀 |
| dist/16.1b6ca79f.js | 1.66kb | +0.00b | 7.97s | -467.00ms 🚀 |
| dist/feedback.d5590b34.js | 1.64kb | +0.00b | 9.06s | -3.55s 🚀 |
| dist/workerHasher.6d5b22d7.js | 1.54kb | +0.00b | 1.02m | +47.91s ⚠️ |
| dist/list-number.c35f9589.js | 1.46kb | +0.00b | 8.67s | -556.00ms 🚀 |
| dist/status.8ab4226b.js | 1.46kb | +0.00b | 8.99s | -496.00ms 🚀 |
| dist/16.9ccb5b28.js | 1.38kb | +0.00b | 9.98s | -51.20s 🚀 |
| dist/code.3dd71280.js | 1.38kb | +0.00b | 8.43s | -700.00ms 🚀 |
| dist/16.19c93285.js | 1.33kb | +0.00b | 7.97s | -467.00ms 🚀 |
| dist/16.c7751e34.js | 1.33kb | +0.00b | 9.71s | -51.46s 🚀 |
| dist/16.51049e2c.js | 1.32kb | +0.00b | 3.83s | -1.29s 🚀 |
| dist/16.18344f47.js | 1.31kb | +0.00b | 8.43s | -577.00ms 🚀 |
| dist/link.2f4427e6.js | 1.30kb | +0.00b | 8.67s | -556.00ms 🚀 |
| dist/heading6.fdbbdf4d.js | 1.30kb | +0.00b | 9.06s | -546.00ms 🚀 |
| dist/heading3.2113b7a1.js | 1.29kb | +0.00b | 9.06s | -546.00ms 🚀 |
| dist/16.2c4c1846.js | 1.27kb | +0.00b | 8.43s | -577.00ms 🚀 |
| dist/emoji.93ae61d3.js | 1.23kb | +0.00b | 8.55s | -573.00ms 🚀 |
| dist/16.d17ab186.js | 1.23kb | +0.00b | 8.23s | -530.00ms 🚀 |
| dist/16.436f097f.js | 1.21kb | +0.00b | 3.83s | -1.29s 🚀 |
| dist/16.b4da0acb.js | 1.19kb | +0.00b | 8.23s | -530.00ms 🚀 |
| dist/16.02c68c4f.js | 1.18kb | +0.00b | 8.43s | -577.00ms 🚀 |
| dist/heading5.e195aab1.js | 1.17kb | +0.00b | 9.06s | -546.00ms 🚀 |
| dist/16.b46ef690.js | 1.14kb | +0.00b | 3.83s | -1.29s 🚀 |
| dist/16.4e3b0235.js | 1.14kb | +0.00b | 8.23s | -531.00ms 🚀 |
| dist/16.9ba230ea.js | 1.13kb | +0.00b | 7.97s | -53.06s 🚀 |
| dist/16.a37df53b.js | 1.13kb | +0.00b | 9.98s | -51.20s 🚀 |
| dist/16.9adb4ec7.js | 1.13kb | +0.00b | 9.98s | +1.22s ⚠️ |
| dist/16.eafe1a11.js | 1.13kb | +0.00b | 7.97s | -467.00ms 🚀 |
| dist/heading2.a2a1be3f.js | 1.11kb | +0.00b | 8.99s | -618.00ms 🚀 |
| dist/16.3920766f.js | 1.10kb | +0.00b | 7.93s | +1.95s ⚠️ |
| dist/component.6600f83e.js | 1.07kb | +0.00b | 7.93s | +1.95s ⚠️ |
| dist/mention.4d64e059.js | 1.07kb | +0.00b | 8.67s | -708.00ms 🚀 |
| dist/heading4.ec980e93.js | 1.06kb | +0.00b | 9.06s | -546.00ms 🚀 |
| dist/16.afcb09b6.js | 1.05kb | +0.00b | 8.23s | -771.00ms 🚀 |
| dist/layout.52ba15f7.js | 1.05kb | +0.00b | 8.67s | -556.00ms 🚀 |
| dist/divider.16defde6.js | 1.03kb | +0.00b | 8.55s | -573.00ms 🚀 |
| dist/quote.aa25b83c.js | 1.02kb | +0.00b | 8.99s | -496.00ms 🚀 |
| dist/action.9aa72b39.js | 1.00kb | +0.00b | 8.43s | -577.00ms 🚀 |
| dist/decision.20244046.js | 1010.00b | +0.00b | 8.55s | -573.00ms 🚀 |
| dist/panel-warning.97a0e3b3.js | 1008.00b | +0.00b | 8.86s | -522.00ms 🚀 |
| dist/list.c9147a62.js | 978.00b | +0.00b | 8.67s | -555.00ms 🚀 |
| dist/heading1.7d899ba5.js | 976.00b | +0.00b | 8.99s | -496.00ms 🚀 |
| dist/panel-error.8d11e8b7.js | 906.00b | +0.00b | 8.86s | -523.00ms 🚀 |
| dist/panel.a28c56ee.js | 897.00b | +0.00b | 8.86s | -623.00ms 🚀 |
| dist/table.5a4874ee.js | 889.00b | +0.00b | 8.99s | -496.00ms 🚀 |
| dist/panel-success.9b923e58.js | 845.00b | +0.00b | 8.86s | -522.00ms 🚀 |
| dist/panel-note.97a4d541.js | 841.00b | +0.00b | 8.86s | -522.00ms 🚀 |
| dist/media-viewer-analytics-error-boundary.0afa7031.js | 755.00b | +0.00b | 17.25s | -950.00ms 🚀 |
| dist/EmojiPickerComponent.2e54b9c6.js | 535.00b | +0.00b | 1.01m | +47.62s ⚠️ |
| dist/simpleHasher.a29b2a8e.js | 528.00b | +0.00b | 1.06m | -43.53s 🚀 |
| dist/index.html | 119.00b | +0.00b | 1.09m | +1.02m ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.fa25440a.js | 2.72mb | +3.00b ⚠️ | 201.00ms | +47.00ms ⚠️ |
| dist/pdfRenderer.15a6a023.js | 1.11mb | +0.00b | 186.00ms | +31.00ms ⚠️ |
| dist/editorView.e06544fb.js | 587.12kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/popup.6c453e3c.js | 169.96kb | +0.00b | 186.00ms | +37.00ms ⚠️ |
| dist/Toolbar.83ac0b6e.js | 92.34kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/EmojiPickerComponent.ae601247.js | 75.29kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/media-viewer.bcba38b3.js | 71.96kb | +0.00b | 186.00ms | +31.00ms ⚠️ |
| dist/card.645a5c17.js | 55.01kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/card.0313bac8.js | 52.78kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/Modal.19ffed32.js | 41.27kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/component.d7c29980.js | 30.70kb | +0.00b | 200.00ms | +46.00ms ⚠️ |
| dist/component.54080a26.js | 22.50kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/DatePicker.0df05dac.js | 21.21kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/EmojiPickerComponent.7dc0ebd0.js | 18.23kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/dropzone.7c704bdb.js | 17.27kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/smartMediaEditor.9623570c.js | 16.87kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/js.5733a017.js | 16.40kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/ui.fee77548.js | 13.97kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/workerHasher.1e706e77.js | 11.75kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.1990d771.js | 8.67kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/component.77c0f682.js | 6.05kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/card.383bd9bd.js | 5.61kb | +0.00b | 186.00ms | +30.00ms ⚠️ |
| dist/media-viewer.59c17dad.js | 3.92kb | +0.00b | 186.00ms | +31.00ms ⚠️ |
| dist/EmojiPickerComponent.627aca6b.js | 3.58kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/index.ac562c80.css | 3.46kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/png-chunks-extract.a844fdb6.js | 3.41kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/dropzone.3c67f0a6.js | 3.23kb | +0.00b | 186.00ms | +37.00ms ⚠️ |
| dist/Modal.0304b9ed.js | 3.08kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/clipboard.da112b6e.js | 2.75kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/16.290b4a43.js | 2.38kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/EmojiUploadComponent.d54a1fc6.js | 2.18kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/card.eea6ad8c.js | 2.03kb | +0.00b | 186.00ms | +37.00ms ⚠️ |
| dist/ResourcedEmojiComponent.abe8596c.js | 2.00kb | +0.00b | 201.00ms | +47.00ms ⚠️ |
| dist/date.231500d2.js | 1.73kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/16.d16f14ed.js | 1.73kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/images.80c92f5e.js | 1.67kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/16.1b6ca79f.js | 1.66kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/feedback.d5590b34.js | 1.64kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/browser.75813db0.js | 1.61kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/workerHasher.6d5b22d7.js | 1.54kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/workerHasher.fa198233.js | 1.54kb | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/list-number.c35f9589.js | 1.46kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/status.8ab4226b.js | 1.46kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/16.9ccb5b28.js | 1.38kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/code.3dd71280.js | 1.38kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/16.19c93285.js | 1.33kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/16.c7751e34.js | 1.33kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/16.51049e2c.js | 1.32kb | +0.00b | 200.00ms | +46.00ms ⚠️ |
| dist/16.18344f47.js | 1.31kb | +0.00b | 188.00ms | +35.00ms ⚠️ |
| dist/link.2f4427e6.js | 1.30kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/heading6.fdbbdf4d.js | 1.30kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/heading3.2113b7a1.js | 1.29kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/16.2c4c1846.js | 1.27kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/emoji.93ae61d3.js | 1.23kb | +0.00b | 188.00ms | +31.00ms ⚠️ |
| dist/16.d17ab186.js | 1.23kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/16.436f097f.js | 1.21kb | +0.00b | 201.00ms | +47.00ms ⚠️ |
| dist/16.b4da0acb.js | 1.19kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/16.02c68c4f.js | 1.18kb | +0.00b | 188.00ms | +35.00ms ⚠️ |
| dist/heading5.e195aab1.js | 1.17kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/expand.450f3aea.js | 1.16kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.2323af57.js | 1.15kb | +0.00b | 186.00ms | +37.00ms ⚠️ |
| dist/16.b46ef690.js | 1.14kb | +0.00b | 201.00ms | +47.00ms ⚠️ |
| dist/16.4e3b0235.js | 1.14kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/16.9ba230ea.js | 1.13kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/16.a37df53b.js | 1.13kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/16.9adb4ec7.js | 1.13kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/16.eafe1a11.js | 1.13kb | +0.00b | 200.00ms | +47.00ms ⚠️ |
| dist/heading2.a2a1be3f.js | 1.11kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/16.3920766f.js | 1.10kb | +0.00b | 201.00ms | +47.00ms ⚠️ |
| dist/component.6600f83e.js | 1.07kb | +0.00b | 201.00ms | +47.00ms ⚠️ |
| dist/mention.4d64e059.js | 1.07kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/heading4.ec980e93.js | 1.06kb | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/16.afcb09b6.js | 1.05kb | +0.00b | 201.00ms | +48.00ms ⚠️ |
| dist/layout.52ba15f7.js | 1.05kb | +0.00b | 188.00ms | +31.00ms ⚠️ |
| dist/divider.16defde6.js | 1.03kb | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/quote.aa25b83c.js | 1.02kb | +0.00b | 188.00ms | +31.00ms ⚠️ |
| dist/action.9aa72b39.js | 1.00kb | +0.00b | 188.00ms | +35.00ms ⚠️ |
| dist/decision.20244046.js | 1010.00b | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/panel-warning.97a0e3b3.js | 1008.00b | +0.00b | 188.00ms | +31.00ms ⚠️ |
| dist/list.c9147a62.js | 978.00b | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/heading1.7d899ba5.js | 976.00b | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/panel-error.8d11e8b7.js | 906.00b | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/panel.a28c56ee.js | 897.00b | +0.00b | 188.00ms | +31.00ms ⚠️ |
| dist/table.5a4874ee.js | 889.00b | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/panel-success.9b923e58.js | 845.00b | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/panel-note.97a4d541.js | 841.00b | +0.00b | 188.00ms | +36.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.f8ddfee3.js | 839.00b | +0.00b | 186.00ms | +37.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.663369b9.js | 835.00b | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.73ba4074.js | 835.00b | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.0afa7031.js | 755.00b | +0.00b | 186.00ms | +36.00ms ⚠️ |
| dist/EmojiPickerComponent.2e54b9c6.js | 535.00b | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/simpleHasher.16c8bf41.js | 528.00b | +0.00b | 187.00ms | +31.00ms ⚠️ |
| dist/simpleHasher.a29b2a8e.js | 528.00b | +0.00b | 185.00ms | +36.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 226.00ms | +64.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/276391866349388300)**

File: packages/core/utils/src/sourcemap.js (hunk 1 above)
Comment (DeMoorJasper) [reply packages/core/utils/src/sourcemap.js:33]: Sure, dont expect file:// in the middle of the filepath but better to be safe I guess

**Comment by DeMoorJasper** [comment]
@mischnic I do not, but apparently it happens as the issue I linked.

Every tool should probably emit relative paths for their sourcemaps though as otherwise it cannot be used cross-machine...

File: packages/core/utils/src/sourcemap.js [hunk 2]
```diff
@@ -30,15 +30,18 @@ export async function loadSourceMapUrl(
   if (match) {
     let url = match[1].trim();
     let dataURLMatch = url.match(DATA_URL_RE);
-    filename = dataURLMatch ? filename : path.join(path.dirname(filename), url);
+    let mapFilePath = dataURLMatch ? filename : url.replace(/^file:\/\//, '');
+    if (!dataURLMatch && !isAbsolute(mapFilePath)) {
+      mapFilePath = path.join(path.dirname(filename), mapFilePath);
+    }
```
Comment (mischnic) [inline packages/core/utils/src/sourcemap.js:36]:
```suggestion
    let mapFilePath;
    if (dataURLMatch) {
        mapFilePath =  isAbsolute(mapFilePath) ? filename : path.join(path.dirname(filename), mapFilePath);
    } else {
        mapFilePath = url.replace(/^file:\/\//, '');
    }
```

File: packages/core/utils/src/sourcemap.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import type {FileSystem} from '@parcel/fs';
 import SourceMap from '@parcel/source-map';
 import path from 'path';
-import {normalizeSeparators} from './path';
+import {normalizeSeparators, isAbsolute} from './path';
```
Comment (mischnic) [inline packages/core/utils/src/sourcemap.js:5]: Not sure if we could use `path.isAbsolute` in this case because e.g. reading from `C:\` on macOS doesn't work anyway?

File: packages/core/utils/src/sourcemap.js (hunk 2 above)
Comment (DeMoorJasper) [reply packages/core/utils/src/sourcemap.js]: Will revert this change, it seems incorrect

Comment (mischnic) [reply packages/core/utils/src/sourcemap.js:36]:
```suggestion
    let mapFilePath;
    if(dataURLMatch) {
        mapFilePath = filename;
    } else {
        mapFilePath = isAbsolute(mapFilePath) ? url.replace(/^file:\/\//, '') : path.join(path.dirname(filename), mapFilePath);
    }
```

**Comment by mischnic** [comment]
Alright, I give up 😬

**Comment by DeMoorJasper** [comment]
@mischnic haha, I'll just clean it up a bit

**Review by mischnic — APPROVED** [review]
